Feature/devops ci (#3799)
conan-io/docs: devops/package_promotions.rst

      

Between those repositories there will be some quality gates. In our case, some packages will be
put in the "testing" repository, for the QA team to test them, for example ``zlib/1.3.1`` and ``openssl/3.2.2``:

.. graphviz::
    :align: center

    digraph repositories {
        node [fillcolor="lightskyblue", style=filled, shape=box]
        rankdir="LR"; 
        subgraph cluster_0 {
            label="Packages server";
            style=filled;
            color=lightgrey;
            subgraph cluster_1 {
                label = "testing\n repository" 
                shape = "box";
                style=filled;
                color=lightblue;
                "zlib/1.3.1";
                "openssl/3.2.2";
            }
 
            subgraph cluster_2 {
                label = "release\n repository" 
                shape = "box";
                style=filled;
                color=lightblue;
                "release" [style=invis];
            } 
            {
                edge[style=invis];
                "zlib/1.3.1" -> "release" ; 
                rankdir="BT";    
            }
        }
    }

> When QA team tests and approves these packages, they can be promoted to the "release" repository.

Reviewer: ```suggestion
When the QA team tests and approves these packages, they can be promoted to the "release" repository.
```

Basically, a promotion is a copy of the packages, including all the artifacts and metadata from the
"testing" to the "release" repository.

> There are different ways to implement and execute a package promotion. Artifactoy has some APIs that can be

Reviewer 2: ```suggestion
There are different ways to implement and execute a package promotion. Artifactory has some APIs that can be
```

used to move individual files or folders. The `Conan extensions repository <https://github.com/conan-io/conan-extensions>`_
contains the ``conan art:promote`` command that can be used to promote Conan "package lists" from one 
server repository to another repository.

If we have a package list ``pkglist.json`` that contains the above ``zlib/1.3.1`` and ``openssl/3.2.2`` packages, then
the command would look like:

.. code-block:: bash
    :caption: Promoting from testing->release

    $ conan art:promote pkglist.json --from=testing --to=release --url=https://<url>/artifactory --user=<user> --password=<password>

> Note that the ``conan art:promote`` doesn't work with ArtifactoryCE, but need pro editions of Artifactory.

Reviewer: ```suggestion
Note that the ``conan art:promote`` command doesn't work with ArtifactoryCE, Pro editions of Artifactory are needed.
```
